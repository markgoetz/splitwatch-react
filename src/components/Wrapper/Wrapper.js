import React from 'react';
import PropTypes from 'prop-types';
import style from './Wrapper.module.css';

export default function Wrapper(props) {
    const baseStyle = style.wrapper;
    let alignStyle = '';

    switch (props.align) {
        case 'left':
            alignStyle = style.left;
            break;
        case 'center':
            alignStyle = style.center;
            break;
        case 'right':
            alignStyle = style.right;
            break;
        default:
            break;
    }

    const wrapperStyle = `${baseStyle} ${alignStyle}`;

    return <div className={wrapperStyle}>{props.children}</div>;
}

Wrapper.propTypes = {
    align: PropTypes.oneOf(['left', 'center', 'right'])
};