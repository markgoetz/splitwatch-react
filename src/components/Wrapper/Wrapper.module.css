.wrapper {
  padding: 0 20px;
}

@media (min-width: 769px) {
  .wrapper {
    width: 600px;
  }

  .center {
    margin: 0 auto;
  }
  
  .left {
    margin: 0 auto 0 0;
  }

  .right {
    margin: 0 0 0 auto;
  }
}