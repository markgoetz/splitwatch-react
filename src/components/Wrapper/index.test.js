import React from 'react';
import Wrapper from './Wrapper';
import { shallow } from 'preact-render-spy';

describe('Wrapper', () => {
  test('Wrapper renders correctly when left-aligned', () => {
    const context = shallow(<Wrapper align="left">
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </Wrapper>);
    expect(context).toMatchSnapshot();
  });

  test('Wrapper renders correctly when center-aligned', () => {
    const context = shallow(<Wrapper align="center">
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </Wrapper>);
    expect(context).toMatchSnapshot();
  });

  test('Wrapper renders correctly when right-aligned', () => {
    const context = shallow(<Wrapper align="right">
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </Wrapper>);
    expect(context).toMatchSnapshot();
  });
});