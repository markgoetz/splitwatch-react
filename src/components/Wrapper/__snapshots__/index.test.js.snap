// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper Wrapper renders correctly when center-aligned 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined undefined">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;

exports[`Wrapper Wrapper renders correctly when left-aligned 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined undefined">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;

exports[`Wrapper Wrapper renders correctly when right-aligned 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined undefined">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;
